src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1315,23 +1304,18 @@ static ucs_status_t uct_ib_mlx5dv_md_open(struct ibv_device *ibv_device,
         goto err;
     }
 
-    md = ucs_calloc(1, sizeof(*md), "ib_mlx5_md");
+    md = ucs_derived_of(uct_ib_md_alloc(sizeof(*md), "ib_mlx5dv_md", ctx),
+                        uct_ib_mlx5_md_t);
```

Comment:
maybe unite uct_ib_md_alloc and uct_ib_md_open_common?

Reply:
the issue is we should do the following actions
```
    uct_ib_mlx5_devx_mr_lru_cleanup(md);
    uct_ib_mlx5_devx_cleanup_flush_mr(md);
    uct_ib_mlx5_md_buf_free(md, md->zero_buf, &md->zero_mem);
```
after `uct_ib_md_close_common()` and before `uct_ib_md_free()` which deallocates PD in `uct_ib_mlx5_devx_md_cleanup()`.

Reply:
can we do these actions before calling uct_ib_md_close_common ?

Reply:
> can we do these actions before calling uct_ib_md_close_common ?

unfortunately, no, because `uct_ib_md_close_common` cleanups rcache, and we should do it prior Memory Regions cleanup

Reply:
ok